### PR TITLE
bug fix and possible fix for #11

### PR DESCRIPTION
**Pull request by JBKahn**
The use of lists there seems to make sense (should be forced as a list, right?).

I'm not sure why you're specifically removing the timezone?

This resolves https://github.com/scrapinghub/dateparser/issues/11 if that was simply a bug. Couldn't tell the intent of the line.

```python
In [1]: from dateparser.date import DateDataParser

In [2]: ddp = DateDataParser()

In [3]: ddp.get_date_data('2014-10-09T17:57:39+00:00')['date_obj']
Out[3]: datetime.datetime(2014, 10, 9, 17, 57, 39, tzinfo=tzutc())

In [4]: ddp.get_date_data('2014-10-09T17:57:39+00:00', '')['date_obj']
Out[4]: datetime.datetime(2014, 10, 9, 17, 57, 39, tzinfo=tzutc())

In [5]: ddp.get_date_data('2014-10-09T17:57:39+00:00', '%Y')['date_obj']
Out[5]: datetime.datetime(2014, 10, 9, 17, 57, 39, tzinfo=tzutc())

```

### CODE REVIEW

**Comment by Allactaga** [comment]
Hi, @JBKahn. Happy New Year! :smiley: 
Are you still planning to work further on this request or can we assign someone to it?

**Comment by JBKahn** [comment]
Yup, I wasn't on my computer a lot over the holidays. I'll pick this up over the next few days

**Comment by Allactaga** [comment]
Great, thanks! :smiley:

**Comment by JBKahn** [comment]
It's been a long time since I've looked over this. Is that the test you were looking for here?
```
----------------------------------------------------------------------
Ran 113 tests in 2.364s
```

**Comment by Allactaga** [comment]
Hi @JBKahn , sorry for taking me so long to respond. Can you please merge `master` branch in yours? We made some major changes there.

**Comment by JBKahn** [comment]
sure

**Comment by JBKahn** [comment]
I'll take another look when I have time, I'm just going to assume the tests won't pass and I'll need to fix something.

**Comment by JBKahn** [comment]
well I'll be damned. I'm better at this merging business than I thought.

File: dateparser/date.py [hunk 1]
```diff
@@ -175,7 +175,7 @@ def _try_given_formats(self):
         if not self.date_formats:
             return
 
-        return parse_with_formats(self._get_translated_date_with_formatting(), self.date_formats)
+        return parse_with_formats(self._get_translated_date_with_formatting(), list(self.date_formats))
```
Comment (Allactaga) [inline dateparser/date.py]: Could you please remind me why do we need to construct list here?

Comment (JBKahn) [inline dateparser/date.py]: It's not garunteed to be an iterable and in the event that it isn't it used
to to something foolish in the end as well as iterate over the string.

Comment (Allactaga) [inline dateparser/date.py]: But how would this change help in such case?
```python
>>> list("qwerty")
['q', 'w', 'e', 'r', 't', 'y']
```

Comment (JBKahn) [inline dateparser/date.py]: Right, I never got around to running a test, should have instead put it in
a list if it's a string. Haven't looked at this in a while

Comment (Allactaga) [inline dateparser/date.py]: We used [this check](https://github.com/scrapinghub/dateparser/blob/master/dateparser/date.py#L222) for similar case. May be it is worth to add this group to `dateparser.utils` as `plain_iterable` or something. What do you think?

Comment (JBKahn) [inline dateparser/date.py]: I could get behind that check, works for me. When I get back on my laptop
I'll make the change and run the tests

**Comment by Allactaga** [comment]
Hi @JBKahn, how are you?
What do you say if we finish with this until someone landed more merge conflicts in master? :)

**Comment by JBKahn** [comment]
I'll do it tonight.

**Comment by JBKahn** [comment]
```bash
----------------------------------------------------------------------
Ran 483 tests in 3.980s

OK (skipped=11)
```

**Comment by Allactaga** [comment]
Good for me, but what do you think: wouldn't it be better to raise an error there? I mean so that someone wouldn't pass a dict or object.

**Comment by JBKahn** [comment]
Sure, that works for me. So long as it's handled gracefully.

**Comment by Allactaga** [comment]
Will you make proper change? Or if you want to, i can make it and merge the branch.

**Comment by JBKahn** [comment]
If you could make the change, that would be great!

**Comment by Allactaga** [comment]
Sure.

**Comment by Allactaga** [comment]
Here are the [changes](https://github.com/scrapinghub/dateparser/compare/2e28df572b6e6e076d070752499244e147e6fce9...679dfba5d40ba85bb59e85ce123287e9de17a55d) i made.

**Pull request merged**